java/arcs/core/entity/Reference.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.entity
+
+import arcs.core.common.Referencable
+import arcs.core.storage.Reference as StorageReference
+
+/** A reference to an [Entity]. */
+class Reference<T : Entity>(
+    val entitySpec: EntitySpec<T>,
+    private val storageReference: StorageReference
+) {
+    /** The schema hash for the [Reference]'s associated schema. */
+    val schemaHash: SchemaHash
+        get() = entitySpec.SCHEMA.hash
```

Comment:
entitySpec is final & immutable, right? this could just be a normal val declaration

Reply:
Good call, done.